server/src/main/java/org/candlepin/resource/StatusResource.java

```diff
@@ -86,22 +83,11 @@ public StatusResource(RulesCurator rulesCurator, Configuration config, JsRunnerP
         if (config == null || !config.getBoolean(ConfigProperties.STANDALONE)) {
             standalone = false;
         }
-
-        try (FileReader reader = new FileReader(ConfigProperties.KEYCLOAK_FILEPATH)) {
-            //Read JSON file
-            ObjectMapper objectMapper = new ObjectMapper();
-            JsonNode rootNode = objectMapper.readTree(reader);
-            realm = rootNode.get("realm").asText();
-            authUrl = rootNode.get("auth-server-url").asText();
-            resource = rootNode.get("resource").asText();
-        }
-        catch (FileNotFoundException e) {
-            log.error("Keycloak.json file not found", e);
-        }
-        catch (IOException e) {
-            log.error("IOException while using Keycloak.json", e);
+        if (keycloakAdapterConfiguration.getAdapterConfig() != null) {
+            realm = keycloakAdapterConfiguration.getAdapterConfig().getRealm();
+            authUrl = keycloakAdapterConfiguration.getAdapterConfig().getAuthServerUrl();
+            resource = keycloakAdapterConfiguration.getAdapterConfig().getResource();
```

Comment:
A lot of repetition here. Grab the config once, reference it locally. This is also more efficient in cases where the getter isn't a simple one-liner the compiler will optimize out. And since the StatusResource object has no relation to the keycloak config, it should be treating all of the keycloak objects as black box.

```
AdapterConfig keycloakConfig = keycloakAdapterConfiguration.getAdapterConfig();
if (keycloakConfig != null) {
    this.realm = keycloakConfig.getRealm();
    this.authUrl = keycloakConfig.getAuthServerUrl();
    this.resource = keycloakConfig.getResource();
}
```

Also, whitespace is your friend. Separate logical blocks with a line of whitespace to separate them from other logic and make it clear at a glance that it is not connected to the previous block.